### PR TITLE
[tests] Import tests/codegen from monodroid/5fc80f27

### PR DESCRIPTION
**Pull request by jonpryor**
`tests/CodeGen-Binding` is a full-stack integration jack for JNI
bindings, ensuring that `generator.exe` and company work as expected.
It is *somewhat* duplicative of the Java.Interop `generator` unit
tests, except these are actually executed, while the `generator` tests
are *not* executed.

`tests/CodeGen-MkBundle` is a test for the "Bundle assemblies into
native code" feature, but isn't part of the "proper" unit test
infrastructure, as it doesn't yet contain an `Instrumentation` type.

TODO: Add an `Instrumentation` type to `tests/CodeGen-MkBundle`.

### CODE REVIEW

**Comment by jonpryor** [comment]
Something is wrong; test execution fails:

```
        Task "Adb"
                Using task Adb from Xamarin.Android.Tools.BootstrapTasks.Adb, Xamarin.Android.Tools.BootstrapTasks, Version=1.0.6178.31998, Culture=neutral, PublicKeyToken=null
                Task Adb
                  Arguments: -s emulator-5600  install -r "/Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/tests/../bin/TestDebug/Xamarin.Android.JcwGen_Tests.apk"
                Tool /Users/builder/android-toolchain/sdk/platform-tools/adb execution started with arguments: -s emulator-5600  install -r "/Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/tests/../bin/TestDebug/Xamarin.Android.JcwGen_Tests.apk" 
                Environment variables being passed to the tool:
                1569 KB/s (62064461 bytes in 38.624s)
                        pkg: /data/local/tmp/Xamarin.Android.JcwGen_Tests.apk
                Failure [INSTALL_PARSE_FAILED_NO_CERTIFICATES]
                Tool /Users/builder/android-toolchain/sdk/platform-tools/adb execution finished.
                  [Output] Output:
                    1569 KB/s (62064461 bytes in 38.624s)
                        pkg: /data/local/tmp/Xamarin.Android.JcwGen_Tests.apk
                    Failure [INSTALL_PARSE_FAILED_NO_CERTIFICATES]
```

**Comment by jonpryor** [comment]
...because `Xamarin.Android.JcwGen-Tests.projitems` referenced the wrong file.

**Comment by jonpryor** [comment]
New tests are now being executed. No additional failures.

File: tests/CodeGen-Binding/Xamarin.Android.JcwGen-Tests/.gitignore~ [hunk 1]
```diff
@@ -0,0 +1,2 @@
+jni/timing.c
```
Comment (radekdoulik) [inline tests/CodeGen-Binding/Xamarin.Android.JcwGen-Tests/.gitignore~]: I think we don't want the backup files (ending with ~) in the repo.

**Pull request merged**